aten/src/ATen/native/sparse/cuda/SoftMax.cu

```diff
@@ -392,8 +392,6 @@ void cuda_sparse_coo_softmax(
   }
 
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();
-  auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
-  auto policy = thrust::cuda::par(allocator).on(stream);
```

Comment:
No, this PR propagates more C++ flags to nvcc so these unused variables fail the Werror build.

Reply:
I'm not an expert here, but aren't `policy` is something like a thread-guard/context manager (i.e. until this class is destroyed all thrust primitives are executed on the cuda stream specified by `stream` variable instead of a default one?
@ngimel can you confirm whether its the case of changes here are ok to remove?
P.S. Can you perhaps move this change into separate PR that I can land ASAP?

Reply:
You can see from other parts of this file that the execution policy is passed explicitly to the function:
https://github.com/pytorch/pytorch/blob/e83d0a606ad3b72e7054ac56f04011226d6bca31/aten/src/ATen/native/sparse/cuda/SoftMax.cu#L244-L245

So, the fact it compiles without this variable proves that it's dead code. You can also see the function doesn't actually use thrust directly, instead it's all in `compute_pool_max` which defines it's own `policy`.

I've moved this change out into  #65539.